src/ssl.c

```diff
@@ -16101,10 +16656,75 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
             ret = WOLFSSL_SUCCESS;
         }
 
+    #ifdef WOLFSSL_DUAL_ALG_CERTS
+        if (ret == WOLFSSL_SUCCESS)
+            /* Set the label for the alternative key, too. User can still
+             * override that afterwards. */
+            ret = wolfSSL_CTX_use_AltPrivateKey_Label(ctx, label, devId);
+    #endif
+
         return ret;
     }
 #endif /* WOLF_PRIVATE_KEY_ID */
 
+#if defined(WOLF_PRIVATE_KEY_ID) && defined(WOLFSSL_DUAL_ALG_CERTS)
+    int wolfSSL_CTX_use_AltPrivateKey_id(WOLFSSL_CTX* ctx,
+                                         const unsigned char* id,
+                                         long sz, int devId, long keySz)
+    {
+        int ret = wolfSSL_CTX_use_AltPrivateKey_Id(ctx, id, sz, devId);
+
+        if (ret == WOLFSSL_SUCCESS)
+            ctx->altPrivateKeySz = (word32)keySz;
+
+        return ret;
+    }
+
+    int wolfSSL_CTX_use_AltPrivateKey_Id(WOLFSSL_CTX* ctx,
+                                         const unsigned char* id,
+                                         long sz, int devId)
+    {
+        int ret = WOLFSSL_FAILURE;
+
+        FreeDer(&ctx->altPrivateKey);
+        if (AllocDer(&ctx->altPrivateKey, (word32)sz, ALT_PRIVATEKEY_TYPE,
+                                                              ctx->heap) == 0) {
+            XMEMCPY(ctx->altPrivateKey->buffer, id, sz);
+            ctx->altPrivateKeyId = 1;
+            if (devId != INVALID_DEVID)
+                ctx->altPrivateKeyDevId = devId;
+            else
+                ctx->altPrivateKeyDevId = ctx->devId;
+
+            ret = WOLFSSL_SUCCESS;
+        }
+
+        return ret;
+    }
+
+    int wolfSSL_CTX_use_AltPrivateKey_Label(WOLFSSL_CTX* ctx, const char* label,
+                                            int devId)
+    {
+        int ret = WOLFSSL_FAILURE;
+        word32 sz = (word32)XSTRLEN(label) + 1;
```

Comment:
NULL sanity check on 'ctx' and 'label' before use with public facing function.

Reply:
Fixed in next commit.